backend/core/utils.py

```
@@ -0,0 +1,20 @@
+    Args:
+        instance: The Pydantic model instance to validate
+
+    Raises:
+        ValidationError: If validation fails
+    """
+    try:
+        instance.model_validate(instance.model_dump())
+    except ValidationError as e:
+        raise ValidationError.from_exception_data(
+            title="Validation Error", line_errors=e.errors(), model=instance.__class__
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)